src/applicationManager.ts

```diff
@@ -372,3 +374,61 @@ export async function addOptionalClaims({
     await errorHandler("Add optional claims", result);
   }
 }
+
+export async function addClientSecret({
+  token,
+  applicationId,
+  clientSecret,
+}: {
+  token: string;
+  applicationId: string;
+  clientSecret: CLIENTSECRET;
+}): Promise<void> {
+  if (clientSecret && clientSecret.key_vault_name) {
+    const application = await readApplication({ token, applicationId });
+
+    if (
+      application.passwordCredentials &&
+      application.passwordCredentials.length > 0 &&
+      areAllPasswordsExpired(clientSecret.name, application.passwordCredentials)
+    ) {
+      const body = {
+        passwordCredential: {
+          displayName: `${clientSecret.name}`,
+        },
+      };
+      const addPasswordResult = await fetch(
+        `https://graph.microsoft.com/v1.0/applications/${applicationId}/addPassword`,
+        {
+          method: "POST",
+          headers: {
+            Authorization: `Bearer ${token}`,
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify(body),
+        }
+      );
+      await errorHandler("Add client password", addPasswordResult);
+
+      const clientPassword = (await addPasswordResult.json()).secretText;
+
+      const credential = new DefaultAzureCredential();
+      const keyVaultName = clientSecret.key_vault_name;
+      const url = "https://" + keyVaultName + ".vault.azure.net";
+      const secretClient = new SecretClient(url, credential);
+      await secretClient.setSecret(clientSecret.name, clientPassword);
+    }
+  }
+}
+
+function areAllPasswordsExpired(name: string, passwordCredentials: Array<any>) {
```

Comment:
why check all of them? aren't you interested in a specific one?

Reply:
I am not deleting the old one when adding a new one. If we delete old one, we might end up breaking the app till the new secret is propagated to application. So, checking all secrets with the display name and if all of them expired/(about to expire), then i am creating a new one.

Reply:
but you are using the user provided name? I guess this works but it depends on the user updating the name here ahead of time? would it be better to generate a suffix to add?

Reply:
we can have multiple with same display name. is that what is ur concern ?

Reply:
ah right yeah is was